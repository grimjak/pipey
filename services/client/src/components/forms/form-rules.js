export const loginFormRules = [
    {
    id: 1,
    field: 'username',
    name: 'Username is required.',
    valid: false
    },
    {
    id: 2,
    field: 'password',
    name: 'Password is required.',
    valid: false
    }
    ];