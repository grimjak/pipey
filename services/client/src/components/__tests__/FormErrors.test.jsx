import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';
import FormErrors from '../forms/FormErrors';
import { loginFormRules } from '../forms/form-rules.js';

const loginFormProps = {
    formType: 'login',
    formRules: loginFormRules
}

test('FormErrors (with login form) renders properly', () => {
    const wrapper = shallow(<FormErrors {...loginFormProps} />);
    const ul = wrapper.find('ul');
    expect(ul.length).toBe(1);
    const li = wrapper.find('li');
    expect(li.length).toBe(2);
    expect(li.get(0).props.children).toContain(
        'Username is required');
    expect(li.get(1).props.children).toContain(
        'Password is required');
});

test('FormErrors (with login form) renders a snapshot properly', () => {
    const tree = renderer.create(
        <Router><FormErrors {...loginFormProps} /></Router>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});