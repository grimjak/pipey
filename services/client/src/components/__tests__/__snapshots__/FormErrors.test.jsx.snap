// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormErrors (with login form) renders a snapshot properly 1`] = `
<div>
  <ul
    className="validation-list"
  >
    <li
      className="error"
    >
      Username is required.
    </li>
    <li
      className="error"
    >
      Password is required.
    </li>
  </ul>
  <br />
</div>
`;
